test/suites/clustering_move.sh

```diff
@@ -0,0 +1,198 @@
+test_clustering_move() {
+  # shellcheck disable=2039,3043,SC2034
+  local LXD_DIR
+
+  setup_clustering_bridge
+  prefix="lxd$$"
+  bridge="${prefix}"
+
+  setup_clustering_netns 1
+  LXD_ONE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_ONE_DIR}"
+  ns1="${prefix}1"
+  spawn_lxd_and_bootstrap_cluster "${ns1}" "${bridge}" "${LXD_ONE_DIR}"
+
+  # Add a newline at the end of each line. YAML as weird rules..
+  cert=$(sed ':a;N;$!ba;s/\n/\n\n/g' "${LXD_ONE_DIR}/cluster.crt")
+
+  # Spawn a second node
+  setup_clustering_netns 2
+  LXD_TWO_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_TWO_DIR}"
+  ns2="${prefix}2"
+  spawn_lxd_and_join_cluster "${ns2}" "${bridge}" "${cert}" 2 1 "${LXD_TWO_DIR}"
+
+  # Spawn a third node
+  setup_clustering_netns 3
+  LXD_THREE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_THREE_DIR}"
+  ns3="${prefix}3"
+  spawn_lxd_and_join_cluster "${ns3}" "${bridge}" "${cert}" 3 1 "${LXD_THREE_DIR}"
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster set node2 scheduler.instance=group
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster set node3 scheduler.instance=manual
+
+  ensure_import_testimage
+
+  # Preparation
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar1
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node1 foobar1,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar2
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node2 foobar2,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar3
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node3 foobar3,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c1 --target node1
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c2 --target node2
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c3 --target node3
+
+  # Perform default move tests falling back to the built in logic of choosing the node
+  # with the least number of instances when targeting a cluster group.
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target node2
+  LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target @foobar1
+  LXD_DIR="${LXD_ONE_DIR}" lxc info c1 | grep -q "Location: node1"
+
+  # c1 cannot be moved to the default group since it is already a member
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target=@default || false
+
+  # Perform standard move tests using the `scheduler.instance` cluster member setting.
+  # At this stage we have:
```

Comment:
This comment through me a bit for 2 reasons:

1. It said "at this stage", which I read to be on the line it was at, and then got confused because on lines 70 and 71 it then proceed to set them up in that way.
2. The comment references "direct" targeting but the term used in the command is "manual", so would be good to be consistent here.